pkg/controllers/state/suite_test.go

```diff
@@ -683,6 +687,28 @@ var _ = Describe("Inflight Nodes", func() {
 			ExpectReconcileSucceeded(ctx, machineController, client.ObjectKeyFromObject(machine))
 			ExpectStateNodeNotFoundForMachine(machine)
 		})
+		It("should ignore node updates if nodes don't have provider id and are owned", func() {
```

Comment:
shouldn't it be something like this as both machine nodes and nodeclaim nodes will have providerIDs?

I'd also add the machine part on the end so it's easier to know why it's necessary (although I just had to expand the file and see it was in the Machine section)
```suggestion
		It("should ignore node updates if nodes have old labels and are owned for machines", func() {
```

Reply:
The case here is that we don't respect node updates if we don't have a provider ID yet since we would effectively have double the capacity for a bit when we expect that node to be matched up with a Machine/NodeClaim very soon. I'm inclined to leave this description as-is.